activerecord/lib/active_record/signed_id.rb

```diff
@@ -52,7 +57,8 @@ module ClassMethods
       def find_signed(signed_id, purpose: nil, on_rotation: nil)
         raise UnknownPrimaryKey.new(self) if primary_key.nil?
 
-        if id = signed_id_verifier.verified(signed_id, purpose: combine_signed_id_purposes(purpose), on_rotation: on_rotation)
+        options = { on_rotation: on_rotation }.compact
+        if id = signed_id_verifier.verified(signed_id, purpose: combine_signed_id_purposes(purpose), **options)
```

Comment:
Why this change?

Reply:
The `Message::Rotator#read_message` method is defined like below:
```ruby
def read_message(message, on_rotation: @on_rotation, **options)
  ...
end
```
and when the caller sets `on_rotation` by `nil` it doesn't use `@on_rotation` value. So I'm using `compact` to filter out `nil` values.

Reply:
That is a bug introduced in #54373, isn't it?  In that case, it should be fixed in a separate PR.

Reply:
Yes and no! Previously, it wasn't possible to set `@on_rotation` for `signed_id_verifier`. In other words, there was no way to write a test that would fail with the previous implementation. After using the centralized `message_verifiers` it can be considered as bug.

Reply:
But apps can specify a custom `signed_id_verifier`, no?  For example:

```ruby
Post.signed_id_verifier = ActiveSupport::MessageVerifier.new(..., on_rotation: callback)
```

Reply:
Yeah that's right. I will try it and if it's the case I'll create a separate PR for it. 👍

Reply:
@jonathanhefner You were right. Here is a separate PR for the fix that you mentioned, please review: https://github.com/rails/rails/pull/54523
🙏